specs/searchable-asn-group/searchable-asn-group.md

```diff
@@ -0,0 +1,112 @@
+# Searchable ASN Filters (ff-asn-group)
+
+Some data sets have huge amounts of filters per filter group (150+). Too many to display all at once and too many to browse by eye.
+
+To improve usability the `ff-asn-group` gets an _optional_ search field. It is scoped to its associated filter group.
+
+
+## General
+
+This feature is an addition to the existing `ff-asn-group`. HTML examples are limited to new parts and minimal context for focus.
+
+The new search field is always the first element inside the collapsible `.ffw-wrapper` container. This means the filter group must be expanded in order to use it.
+
+Users may specify a custom search field template. It is defined via the `slot="filterSearch"` attribute. As it is a slot, its location inside the `ff-asn-group` template is irrelevant.
+
+## Features
+
+#### search as you type
+
+Filter items are searched for while user is typing in search field. While search term is present, filter group is kind of in _"search mode"_ compared to _"regular mode"_.
+
+- matching with `new Regexp(searchTerm, "gi")` or better algorithm
+- while search term is present, _"show more"_ disappears, matches of detailed and hidden links are merged into one list
+- not matching items are hidden
+- order of matched results remains unchanged compared to order in original FF response
+- no special treatment for already selected filters
+  - selected filters are filtered just like all others
+
+If a filter item (selected or unselected) is clicked while in _"search mode"_, the search field is cleared and the component returns to _"regular mode"_.
+
+
+## Default HTML
+
+The search field is only rendered if the attribute `searchable-from` is set with a valid number on `ff-asn-group`.
+
+If `searchable-from` is not set, the search field will not be rendered.
+
+#### Setup
+```html
+<ff-asn-group searchable-from="31">
+</ff-asn-group>
+```
+
+#### Rendered
+
+Assuming there are `31` or more links (_detailed_ and _hidden_ combined) in this group.
+
+```html
+<ff-asn-group searchable-from="31">
+    <div class="ffw-asn-group-container">
+        <div class="ffw-wrapper">
+
+            <div slot="filterSearch"><input></div>  <!-- this is the new part -->
+
+            <div data-container="detailedLinks">
+            ...
+```
+
+_Note that the search field is always the first item inside `.ffw-wrapper`._
+
+
+## User specified templates
+
+User templates are specified through the `slot="filterSearch"` attribute.
+
+The `input` element is mandatory. An exception is thrown if it is not provided.
+
+#### Setup
+
+```html
+<ff-asn-group for-group="Manufacturer" searchable-from="31">
+    <div slot="filterSearch">
+        <i class="magnifying-glass"></i>
+        <input placeholder="Search {{group.name}}...">
```

Comment:
Also assume that developers might want to use a `<input type="search">`. There you can clear the field with an **x**. Maybe this action needs special treatment to return to _"regular mode"_.